Dockerfile

```diff
@@ -1,19 +1,34 @@
 # Build the SPIFFE CSI Driver binary
-FROM golang:1.19.3-alpine AS builder
+FROM --platform=${BUILDPLATFORM} golang:1.19.3-alpine AS base
 ARG GIT_TAG
 ARG GIT_COMMIT
 ARG GIT_DIRTY
 WORKDIR /code
 RUN apk --no-cache --update add make
 COPY go.* ./
-RUN go mod download
+RUN --mount=type=cache,target=/go/pkg/mod go mod download
 COPY . .
+
+# xx is a helper for cross-compilation
+# when bumping to a new version analyze the new version for security issues
+# then use crane to lookup the digest of that version so we are immutable
+# crane digest tonistiigi/xx:1.1.2
+FROM --platform=${BUILDPLATFORM} tonistiigi/xx@sha256:9dde7edeb9e4a957ce78be9f8c0fbabe0129bf5126933cd3574888f443731cda AS xx
+
+FROM --platform=${BUILDPLATFORM} base as builder
+ARG TARGETPLATFORM
+ARG TARGETARCH
 ENV CGO_ENABLED=0
-RUN make GIT_TAG="${GIT_TAG}" GIT_COMMIT="${GIT_COMMIT}" GIT_DIRTY="${GIT_DIRTY}" build
+COPY --link --from=xx / /
+RUN xx-go --wrap
+RUN --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/go/pkg/mod \
+    if [ "$TARGETARCH" = "arm64" ]; then CC=aarch64-alpine-linux-musl; fi && \
```

Comment:
We don't need this, correct? we don't use cgo so overriding CC for this arch isn't necessary.